http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/Queries.scala

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import cats.data.Chain
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.ListClusterResponse
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, DataprocInfo}
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterStatus, IP, OperationName}
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+
+import scala.concurrent.ExecutionContext
+
+object Queries {
+  type ClusterJoinLabelJoinRuntimeConfig = Query[((ClusterTable, Rep[Option[LabelTable]]), Rep[Option[RuntimeConfigTable]]), ((ClusterRecord, Option[LabelRecord]), Option[RuntimeConfigRecord]), Seq]
+
+  val runtimeConfigs = TableQuery[RuntimeConfigTable]
+
+  val clusterLabelRuntimeConfigQuery: ClusterJoinLabelJoinRuntimeConfig = {
+    for {
+      (cluster, label) <- clusterQuery
+        .joinLeft(labelQuery).on(_.id === _.clusterId)
+        .joinLeft(runtimeConfigs).on(_._1.runtimeConfigId === _.id)
+    } yield (cluster, label)
+  }
+
+  /**
+    * return DB generated id
+   */
+  def upsertRuntime(runtimeConfig: RuntimeConfig, dateAccessed: Instant): DBIO[RuntimeConfigId] =  {
+    (runtimeConfigs.returning(runtimeConfigs.map(_.id)) += RuntimeConfigRecord(RuntimeConfigId(0), runtimeConfig, dateAccessed))
```

Comment:
Does this upsert? As opposed to slick `.insertOrUpdate`

Reply:
oh yea...this should use `insertOrUpate`